src/lib/journey/journey.store.ts

```diff
@@ -3,13 +3,61 @@ import type { StepOptions } from '@forgerock/javascript-sdk/lib/auth/interfaces'
 import { writable, type Writable } from 'svelte/store';
 
 import { htmlDecode } from '$journey/_utilities/decode.utilities';
-import type { JourneyStore, JourneyStoreValue, StepTypes } from './journey.interfaces';
+import type { JourneyStore, JourneyStoreValue, StackStore, StepTypes } from './journey.interfaces';
 import { interpolate } from '$lib/_utilities/i18n.utilities';
 import {
   authIdTimeoutErrorCode,
   shouldPopulateWithPreviousCallbacks,
 } from './_utilities/step.utilities';
 
+function initializeStack(initOptions?: StepOptions) {
```

Comment:
This is the "stack" to provide the ability to "go back" to a previous journey. Since we can't rely on URL driven state (we don't control the URL), we need an internal history.

Reply:
Should this internal history persist between page reloads? It doesn't and I don't think it should, but want to check the thinking behind this.

Reply:
This is a really good question and one I've been pondering for a while now. When using the modal form factor, I'd say it's pretty non-standard to save the internal state of what's inside the modal throughout a refresh. But, if one is using the inline form factor, maybe?

If they're using the inline form factor, I'd recommend that the devs render the links within the scope of their own app, which can correspond much better with navigations and preserving back button behavior, leaving just the form itself to the Widget. But, there's a hundred ways to build an app.

With that said, I don't think at this moment we should persist the state, but let's collect feedback after our initial release. If we should persist the state, we can have the discussion then. They way it's current built, I don't think adding this after the fact would be difficult.

Let me know what you all think.